app/services/store_sp_metadata_in_session.rb

```diff
@@ -5,6 +5,8 @@ def initialize(session:, request_id:)
   end
 
   def call
+    return unless sp_request
```

Comment:
I take that back. That would not solve the issue where a user deliberately changes the request ID in the URL for the password creation page. It would need to be a new `RequestIdValidator` class called in the `edit` action of `Users::PasswordsController`